docs/DeveloperGuide.md

```diff
@@ -1,11 +1,308 @@
+
+## Getting Started
+### Prerequisites
+1. JDK 11.
+1. IntelliJ IDEA.
+
+### Setting Up
+1. Use a Git tool to fork this repository, or download the .zip file from GitHub and extract the contents into a new folder.
+1. Right-Click on the folder and select “Open folder as Intellij IDEA Community Edition Project”.
+1. Ensure JDK 11 is selected for Gradle, by navigating to Configure > Structure for New Projects > Project Settings > Project > Project SDK.
+
+### Running the Program
+This program can be run once it is compiled. If you have built its artifacts (.jar) file, you may run it using java -jar <filename.jar> on your command line.
+
+## Design
+This section seeks to explain the high-level design of the application. Given below is a quick overview of each component and the explanation of the design architecture in greater detail.
+NotUS is the main class of the application, and handles the initializing and execution of the appropriate classes.
+### Architecture
+![Figure X: Architecture Diagram](./image/developerguide/architecturediagram.png "Architecture Diagram UML")  
+The **Architecture Diagram** shown above describes the high level association operations of the application. 
+A quick overview of the components is as follows:
+
+`Main` is single-class component of `SCRUMptious`. It is responsible for:
+    1. At app launch: Initializes the components in the correct sequence(Storage, UI, Parser), and links them together where appropriate.
+    1. At program exit: Invokes Storage component to save all unsaved data.
+
+The other packages are described below: 
+1. `UI` : The user interface of the app, reads user input and is visible to the user.
+1. `Parser Manager` : The module that reads user inputs, and creates a suitable parser based on the command to make 
+ sense of user input. Respective parser then tells the Command module what to execute.
+1. `ProjectManager` : Manages and stores all the projects added by the user, keeps track of selected project.    
+1. `SprintManager` : Stores all sprints associated with a project.
+1. `TaskManager` : Stores all tasks in backlog associated with a project.
+1. `ProjectMembers` : Stores all team-members associated to a project. 
+1. `Parser` : Creates a suitable parser, based on the command to make sense of the user input. Respective parser then
+ make use the information and call respective commands. 
+ 
+Each of the modules listed above are a collection of constituent classes, with each handling specialized tasks in-line with the SLAP principle.
+
+### UI Component
+![Figure X: Simplified class diagram for UI Component](./image/developerguide/UI.png "User Interface")  
+### Logic Component
+![Figure X: Simplified class diagram for Logic Component](./image/developerguide/parserandcommand.png)
```

Comment:
![image](https://user-images.githubusercontent.com/52351742/97386853-beba0d00-190f-11eb-8bad-1ef637bd9e6a.png)

Should a {abstract} be appended to the end of the method declaration for execute() since it seems to be an abstract method

Reply:
> ![image](https://user-images.githubusercontent.com/52351742/97386853-beba0d00-190f-11eb-8bad-1ef637bd9e6a.png)
> 
> Should a {abstract} be appended to the end of the method declaration for execute() since it seems to be an abstract method